### PR TITLE
Cleanup: Remove unused uu_pname code

### PR DESCRIPTION
**Pull request by ryao**
### Motivation and Context
Coverity caught a possible NULL pointer dereference in dead code. We can delete it all.

### Description
We delete all of the dead code.

### How Has This Been Tested?
I did a local build test.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Review by behlendorf — APPROVED** [review]

File: lib/libuutil/uu_pname.c [hunk 1]
```diff
@@ -88,11 +88,17 @@ uu_warn_internal(int err, const char *format, va_list alist)
 	if (pname != NULL)
 		(void) fprintf(stderr, "%s: ", pname);
 
-	if (format != NULL)
+	if (format != NULL) {
 		(void) vfprintf(stderr, format, alist);
 
-	if (strrchr(format, '\n') == NULL)
```
Comment (tuxoko) [inline lib/libuutil/uu_pname.c:94]: This function is quite weird, it only print strerror if format doesn't have '\n'?
Never the less, why not just change this to `if (!format || strrchr(format, '\n') == NULL)`?
That way we would also print strerror when format is NULL.

Comment (behlendorf) [reply lib/libuutil/uu_pname.c:94]: It appears to be entirely dead code as well.  We should simply remove `uu_pname.[ch]` and all of functions it provides.  None of them are used.  It wouldn't hurt to make a pass over the other `uu_` sources and see how much is really needed.

Comment (ryao) [reply lib/libuutil/uu_pname.c:94]: I had not realized that. I will make another pass over this to delete as much as I can. Deleting dead code is fun. :)

Comment (ryao) [reply lib/libuutil/uu_pname.c:94]: I just pushed a revision that deletes all of this.

**Comment by ryao** [comment]
I updated this to delete all of the dead code rather than fix it. It is a bit of a shame since this library is meant to be shared with other system components, but the only other consumer I know is SMF on Illumos/OpenSolaris, so there are no known consumers of it on Linux or FreeBSD. Deleting this code also diverges from the version in FreeBSD ports:

https://www.freshports.org/devel/libuutil/

**Review by tuxoko — APPROVED** [review]

**Review by behlendorf — APPROVED** [review]

**Comment by behlendorf** [comment]
A quick look suggests we're only using the `uu_list*` and `uu_avl*` interfaces, and not very much.  With a little refactoring it might be possible to remove the need for this library entirely, which wouldn't be such a bad thing.

**Pull request merged**